expression/builtin_time.go

```diff
@@ -3528,6 +3529,18 @@ func (c *unixTimestampFunctionClass) getFunction(ctx sessionctx.Context, args []
 					retDecimal = len(tmpStr) - dotIdx - 1
 				}
 			}
+			if sf, ok := args[0].(*ScalarFunction); ok {
```

Comment:
Can we combine this branch with the above branch for `Constant`?

Reply:
How about
```
			tmpStr := ""
			if cnst, ok := args[0].(*Constant); ok {
				s, _, err := cnst.EvalString(ctx, chunk.Row{})
				if err != nil {
					return nil, err
				}
				tmpStr = s
			} else if sf, ok := args[0].(*ScalarFunction); ok && sf.FuncName.L == ast.GetVar {
				s, _, err := sf.EvalString(ctx, chunk.Row{})
				if err != nil {
					return nil, err
				}
				tmpStr = s
			}
			if tmpStr != "" {
				retDecimal = 0
				if dotIdx := strings.LastIndex(tmpStr, "."); dotIdx >= 0 {
					retDecimal = len(tmpStr) - dotIdx - 1
				}
			}
```